### PR TITLE
Build XcodeSDKs on CI

### PR DESCRIPTION
**Pull request by externl**
This PR adds support for building the Ice C++ Xcode SDKs on with our CI action.

I still need to fix the build, but I just want to first check that this CI setup works. Though I'll wait until https://github.com/zeroc-ice/ice/pull/1774 is merged.

### CODE REVIEW

File: python/python/IceBox/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Generated by slice2py - DO NOT EDIT!
```
Comment (pepone) [inline python/python/IceBox/__init__.py:1]: We need to add a .gitignore to avoid commit this file.

File: cpp/src/Ice/ios/StreamEndpointI.h [hunk 2]
```diff
@@ -35,7 +35,8 @@ class Instance : public IceInternal::ProtocolInstance
 public:
 
     Instance(const Ice::CommunicatorPtr&, Ice::Short, const std::string&, bool);
-    virtual ~Instance();
+    Instance(const Instance&, const IceInternal::ProtocolInstancePtr&);
```
Comment (pepone) [inline cpp/src/Ice/ios/StreamEndpointI.h:38]: I would change `Instance&` to `InstancePtr&` and replace `*_instance.get()` with `_instance`

**Review by pepone — APPROVED** [review]

File: cpp/src/Ice/ios/StreamEndpointI.h (hunk 2 above)
Comment (externl) [reply cpp/src/Ice/ios/StreamEndpointI.h:38]: I was trying to avoid having to forward declare Instance. I guess it could take `shared_ptr<Instance>`?

Comment (externl) [reply cpp/src/Ice/ios/StreamEndpointI.h]: I added the forwards decl.

**Review by pepone — APPROVED** [review]

**Pull request merged**